File: src/main/java/com/epam/eng/pattern/factory/Person.java
package com.epam.eng.pattern.factory;

public class Person {
    private int age;
    private String name;
    private boolean isMale;

    public Person(int age, String name, boolean isMale) {
        this.age = age;
        this.name = name;
        this.isMale = isMale;
    }

    public int getAge() {
        return age;
    }

    public String getName() {
        return name;
    }

    public boolean isMale() {
        return isMale;
    }

    public void eat(String food) {
        System.out.println(name + " is eating " + food);
    }

    public void sleep() {
        System.out.println(name + " is sleeping");
    }

    public void work() {
        System.out.println(name + " is working");
    }

    public void play(String game) {
        System.out.println(name + " is playing " + game);
    }

    public static void main(String args[]) {
        Person person = new Person(25, "John", true);
        person.eat("Pizza");
        person.sleep();
        person.work();
        person.play("Chess");
    }
}

File: ts/factory/Person.ts
class Person {
  private age: number;
  private name: string;
  private isMale: boolean;

  constructor(age: number, name: string, isMale: boolean) {
    this.age = age;
    this.name = name;
    this.isMale = isMale;
  }

  getAge(): number {
    return this.age;
  }

  getName(): string {
    return this.name;
  }

  getIsMale(): boolean {
    return this.isMale;
  }

  eat(food: string): void {
    console.log(`${this.name} is eating ${food}`);
  }

  sleep(): void {
    console.log(`${this.name} is sleeping`);
  }

  work(): void {
    console.log(`${this.name} is working`);
  }

  play(game: string): void {
    console.log(`${this.name} is playing ${game}`);
  }
}

class PersonFactory {
  static createJohn(): Person {
    return new Person(25, "John", true);
  }
}

const john = PersonFactory.createJohn();
john.eat("Pizza");
john.sleep();
john.work();
john.play("Chess");
